stablehlo/dialect/TypeInference.cpp

```diff
@@ -1281,6 +1281,17 @@ LogicalResult inferSelectOp(
                                               inferredReturnShapes);
 }
 
+LogicalResult inferSelectAndScatterOp(
+    Value operend, SmallVectorImpl<Type>& inferredReturnTypes) {
+  Type elementTy = getElementTypeOrSelf(operend.getType());
+  auto rankedTy = operend.getType().dyn_cast<RankedTensorType>();
+  if (rankedTy)
+    inferredReturnTypes.emplace_back(operend.getType());
+  else
+    inferredReturnTypes.emplace_back(UnrankedTensorType::get(elementTy));
```

Comment:
Same feedback as in #404. This function can be simplified to `inferredReturnTypes.push_back(operand.getType())`.